packages/cli/lib/lib/webpack/create-load-manifest.js

```diff
@@ -1,74 +1,67 @@
-const isESM = filename => /\.esm\.js$/.test(filename);
-const isMatch = (filename, condition) => isESM(filename) === condition;
```

Comment:
In case this raises an eyebrow, the asset manifest will always map to ESM output if enabled, CJS if it's not. We don't need to deal with that ourselves anymore.

There's a new test to ensure the `push-manifest.json` is correct and referring to ESM scripts when ESM is enabled.